scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -211,22 +215,33 @@
 	protected ReassemblingHandshakeMessage reassembledMessage;
 
 	/** The handshaker's private key. */
-	protected PrivateKey privateKey;
+	protected final PrivateKey privateKey;
 
 	/** The handshaker's public key. */
-	protected PublicKey publicKey;
+	protected final PublicKey publicKey;
```

Comment:
If we want to keep api compatibility I think that we are not allowed to change this to final :thinking:

(This is where tooling could help)

Reply:
If all classes and all functions are considered to be API, then it will get hard to change something at all. 
"Protected" limit the scope to sub-classes and I don't see, that the API offers a possibility to inject a custom handshaker from outside. Or do I miss something?

> (This is where tooling could help 😅 )

If you consider everything to be the API, then maybe. Otherwise it will be the configuration of that tool, which maybe difficult. I saw your proposal on issue #1159, just create a PR to check, how it works.

Reply:
Maybe we should clarify API question before.

Reply:
:-).

You already started a issue. Maybe you provide the changes to use that plugins. 
Then we will see, when a API break happens. And hopefully, if we agree, we can exclude "case by case" some stuff from the API check of the plugin. 

Do you modify the `publicKey`? 
Do you have a use-case, which makes use of modifying the `publicKey`?

Reply:
>  Maybe you provide the changes to use that plugins

I will try to find time to look at this.